src/Runner.Listener/MessageListener.cs

```diff
@@ -150,6 +150,26 @@ public async Task<Boolean> CreateSessionAsync(CancellationToken token)
                     Trace.Error("Catch exception during create session.");
                     Trace.Error(ex);
 
+                    if (ex is VssUnauthorizedException && creds.Federated is VssOAuthCredential vssOAuthCred)
+                    {
+                        // Check whether we get 401 because the runner registration already removed by the service.
+                        // If the runner registration get deleted, we can't exchange oauth token.
+                        Trace.Error("Test oauth app registration.");
+                        var oauthTokenProvider = new VssOAuthTokenProvider(vssOAuthCred, new Uri(serverUrl));
+                        try
+                        {
+                            await oauthTokenProvider.ValidateCredentialAsync(token);
+                        }
+                        catch (VssServiceResponseException tokenEx)
+                        {
+                            if (tokenEx.Message.Contains("Client registration not found"))
+                            {
+                                _term.WriteError($"Failed to create a session. The runner registration has been removed from the service side, please re-configure.");
```

Comment:
nit: it's fine the way you have it. also, consider: Failed to create a session. The runner registration has been deleted from the server, please re-configure.